src/Aspire.Dashboard/Components/Dialogs/TextVisualizerDialog.razor.cs

```diff
@@ -0,0 +1,230 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using System.Xml;
+using System.Xml.Linq;
+using Aspire.Dashboard.Model.Otlp;
+using Microsoft.AspNetCore.Components;
+using Microsoft.FluentUI.AspNetCore.Components;
+using Microsoft.JSInterop;
+
+namespace Aspire.Dashboard.Components.Dialogs;
+
+public partial class TextVisualizerDialog : ComponentBase, IAsyncDisposable
+{
+    public const string XmlFormat = "xml";
+    public const string JsonFormat = "json";
+    public const string PlaintextFormat = "plaintext";
+
+    private readonly string _copyButtonId = $"copy-{Guid.NewGuid():N}";
+    private readonly string _openSelectFormatButtonId = $"select-format-{Guid.NewGuid():N}";
+
+    private IJSObjectReference? _jsModule;
+    private List<SelectViewModel<string>> _options = null!;
+
+    public HashSet<string?> EnabledOptions { get; } = [];
+    public string FormattedText { get; private set; } = string.Empty;
+    public string FormatKind { get; private set; } = PlaintextFormat;
+
+    [Inject]
+    public required IJSRuntime JS { get; init; }
+
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        if (firstRender)
+        {
+            _jsModule = await JS.InvokeAsync<IJSObjectReference>("import", "/Components/Dialogs/TextVisualizerDialog.razor.js");
+        }
+
+        if (_jsModule is not null && FormatKind != PlaintextFormat)
+        {
+            await _jsModule.InvokeVoidAsync("connectObserver");
+        }
+    }
+
+    protected override void OnParametersSet()
+    {
+        EnabledOptions.Clear();
+        EnabledOptions.Add(PlaintextFormat);
+
+        _options = [
+            new SelectViewModel<string> { Id = PlaintextFormat, Name = Loc[nameof(Resources.Dialogs.TextVisualizerDialogPlaintextFormat)] },
+            new SelectViewModel<string> { Id = JsonFormat, Name = Loc[nameof(Resources.Dialogs.TextVisualizerDialogJsonFormat)] },
+            new SelectViewModel<string> { Id = XmlFormat, Name = Loc[nameof(Resources.Dialogs.TextVisualizerDialogXmlFormat)] }
+        ];
+
+        if (TryFormatJson())
+        {
+            EnabledOptions.Add(JsonFormat);
+        }
+        else if (TryFormatXml())
+        {
+            EnabledOptions.Add(XmlFormat);
+        }
+        else
+        {
+            FormattedText = Content.Text;
+            FormatKind = PlaintextFormat;
+        }
+    }
+
+    private string GetLogContentClass()
+    {
+        return $"log-content highlight-line language-{FormatKind}";
+    }
+
+    private ICollection<StringLogLine> GetLines()
+    {
+        var lines = Regex.Split(FormattedText, Environment.NewLine, RegexOptions.Compiled).ToList();
```

Comment:
`Environment.NewLine` changes based on your environment (unsurprisingly). What about unexpected line endings?

I think formatted data will always use `Environment.NewLine` but unformatted data could have anything.

This is what AI generated when I asked it to split and handle different line endings:

```cs
public static string[] SplitIntoLines(string text)
{
    if (string.IsNullOrEmpty(text))
        return Array.Empty<string>();

    return text.Split(new[] { "\r\n", "\r", "\n" }, StringSplitOptions.None);
}
```

Should have unit tests.

Reply:
done